### PR TITLE
fix(runner): some minor fixes and improvements

### PR DESCRIPTION
**Pull request by ihexxa**
### Changes
- [x] Settings' layout sometimes wrapped.
- [x] Save last settings (iteration, delay, and data)
- [x] Hide the alert modal and display error message in the console

Ref: INS-4366

### CODE REVIEW

File: packages/insomnia/src/ui/routes/runner.tsx [hunk 1]
```diff
@@ -37,57 +36,93 @@ const inputStyle = 'placeholder:italic py-0.5 mr-1.5 px-1 w-24 rounded-sm border
...
+          {
+            value: `------ Start of request (${respInfo.originalRequestName}) ------\n\n\n`,
+            name: 'Text',
+            timestamp: Date.now(),
+          },
+          ...timeline,
+        ];
+      } else {
+        console.error(`failed to read response for the request ${respInfo.originalRequestName}`);
+      }
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/runner.tsx]: I think it would be better here to show something in the timeline that the response failed to be read instead of logging to console. So we don't have to look in two places.

Also since its only a finite number of elements it would be easier to read and debug, and more consistent with the codebase if it was written as a map reduce style rather than a forloop.

File: packages/insomnia/src/ui/routes/runner.tsx [hunk 2]
```diff
@@ -37,57 +36,93 @@ const inputStyle = 'placeholder:italic py-0.5 mr-1.5 px-1 w-24 rounded-sm border
...
+function getTempRunnerSettings(): RunnerSettings | undefined {
+  return tempRunnerSettings;
+}
+
+function updateTempRunnerSettings(settings: Partial<RunnerSettings>) {
+  tempRunnerSettings = {
+    ...tempRunnerSettings,
+    ...settings,
+  };
+}
```
Comment (jackkav) [inline packages/insomnia/src/ui/routes/runner.tsx]: why extract these into functions? it increases indirection from what I can see here. It also caused me some confusion as to how it relates to react state. I'd suggest flattening this abstraction and if possible eliminating or documenting why you need an additional state cache outside of react state.

File: packages/insomnia/src/ui/routes/runner.tsx (hunk 1 above)
Comment (ihexxa) [reply packages/insomnia/src/ui/routes/runner.tsx]: Good idea and logged error to the console. The forloop is because of an async operation inside it.

File: packages/insomnia/src/ui/routes/runner.tsx (hunk 2 above)
Comment (ihexxa) [reply packages/insomnia/src/ui/routes/runner.tsx]: Originally they are for readability, removed and added a comment for it. It basically expects to access the settings between mountings and is a quick approach as may be refactored soon.

File: packages/insomnia/src/ui/routes/runner.tsx (hunk 1 above)
Comment (jackkav) [reply packages/insomnia/src/ui/routes/runner.tsx]: since the async operation is happening for every item in the list you can write a simple map and assign it to a variable first.

**Review by jackkav — DISMISSED** [review]
I'd prefer not to use for var i loops. If you have some more energy for it I'd like you to take another pass, if not you can merge it but it will be inconsistent. 
In one of the review comments you said you have plans to refactor this soon, in future please link to the ticket or the work you propose to refactor it as part of. The idea here is that when we scope things out we aren't dismissing the comment entirely just pushing it into a different PR.

TLDR: use functional style where possible, unless processing a list thousands of items.

File: packages/insomnia/src/ui/routes/runner.tsx (hunk 1 above)
Comment (ihexxa) [reply packages/insomnia/src/ui/routes/runner.tsx]: I tried several ways today and mapping to promises may require some logic handling errors and ordering. Will revisit it when I find a solution in the repo.

**Comment by ihexxa** [comment]
Update: The "reset" button is removed according to the request.
And the ticket for tracking: INS-4374

**Review by CurryYangxx — APPROVED** [review]
